ext/storage_sources/local_store/local_store.c

```diff
@@ -1083,16 +1019,21 @@ local_size(WT_FILE_SYSTEM *file_system, WT_SESSION *session, const char *name, w
 
     local->op_count++;
 
-    /* If the file exists directly in the file system, it's not yet flushed, so use it */
-    ret = stat(name, &sb);
+    /*
+     * We check to see if the file exists in the cache first, and if not the bucket directory. This
+     * maps what a real cloud implementation would do. This will allow us to instrument this code to
+     * try out and measure caching implementations.
+     */
+    if ((ret = local_cache_path(file_system, name, &path)) != 0)
+        goto err;
+
+    ret = stat(path, &sb);
     if (ret == ENOENT) {
-        /* Otherwise, we'll see if it's in the cache directory. */
-        if ((ret = local_cache_path(file_system, name, &path)) != 0)
+        /* It's not in the cache, try the bucket directory. */
+        free(path);
+        if ((ret = local_bucket_path(file_system, name, &path)) != 0)
             goto err;
-
-        ret = stat(path, &sb);
-        /* TODO: tiered: if we still get an ENOENT, then we'd need to ping the cloud to get the
-         * size. */
+        ret = stat(name, &sb);
```

Comment:
It looks like this code is the same as in `local_exist()`. Does it makes sense to refactor into a separate function?